src/javascript/_common/base/socket_base.js

```diff
@@ -145,7 +146,11 @@ const BinarySocketBase = (() => {
                     });
                 }
                 promise_obj.resolve(response);
-                return promise_obj.promise;
+                if (has_callback) {
+                    options.callback(response);
+                } else {
```

Comment:
Shouldn't we return the promise when there is a callback?

Reply:
If there is a callback, we return the `promise` at the end of the `func` on L200.

Reply:
I see. thanks.